src/gdswzd_mod.f90

```diff
@@ -526,104 +684,75 @@ SUBROUTINE GDSWZD_1D_ARRAY(IGDTNUM,IGDTMPL,IGDTLEN,IOPT,NPTS,FILL, &
   
   END SUBROUTINE GDSWZD_1D_ARRAY
 
+  !> Decodes the grib grid description section and
+  !! returns one of the following (for 1-d arrays):
+  !! - iopt=0 Grid and earth coordinates of all grid points.
+  !! - iopt=+1 Earth coordinates of selected grid coordinates.
+  !! - iopt=-1 Grid coordinates of selected earth coordinates.
+  !!
+  !! If the selected coordinates are more than one gridpoint
+  !! beyond the the edges of the grid domain, then the relevant
+  !! output elements are set to fill values.  Also if iopt=0,
+  !! if the number of grid points exceeds the number allotted,
+  !! then all the output elements are set to fill values.
+  !! 
+  !! The actual number of valid points computed is returned too.
+  !!
+  !! Optionally, the vector rotations, map jacobians and
+  !! grid box areas may be returned.
+  !!
+  !! To compute the vector rotations, the optional arguments 'srot' and 'crot'
+  !! must be present. To compute the map jacobians, the
+  !! optional arguments 'xlon', 'xlat', 'ylon', 'ylat' must be present.
+  !!
+  !! To compute the grid box areas, the optional argument
+  !! 'area' must be present.
+  !!
+  !! The current code recognizes the following projections:
+  !! - kgds(1)=000 Equidistant Cylindrical
+  !! - kgds(1)=001 Mercator Cylindrical
+  !! - kgds(1)=003 lambert Conformal Conical
+  !! - kgds(1)=004 Gaussian Cylindrical
+  !! - kgds(1)=005 Polar Stereographic azimuthal
+  !! - kgds(1)=203 E-staggered Rotated Equidistant Cylindrical
+  !! - kgds(1)=205 B-staggered Rotated Equidistant Cylindrical
+  !!
+  !! @param[in] kgds GDS parameters as decoded by w3fi63.  
+  !! @param[in] iopt Option flag.
+  !! - 0 Earth coords of all the grid points.
+  !! - 1 Earth coords of selected grid coords.
+  !! - -1 Grid coords of selected earth coords
+  !!
+  !! @param[in] npts Maximum number of coordinates.
+  !! @param[in] fill Fill value to set invalid output data.
+  !!      Must be impossible value; suggested value: -9999.
+  !! @param[inout] xpts Grid x point coordinates.
+  !! @param[inout] ypts Grid y point coordinates.
+  !! @param[inout] rlon Earth longitudes in degrees E.
+  !! (Acceptable range: -360. to 360.)
+  !!
+  !! @param[inout] rlat Earth latitudes in degrees N.
+  !! (Acceptable range: -90. to 90.)
+  !!
+  !! @param[out] nret Number of valid points computed.
+  !! @param[out] crot Clockwise vector rotation cosines.
+  !! @param[out] srot Clockwise vector rotation sines.
+  !! ugrid=crot*uearth-srot*vearth;
+  !! vgrid=srot*uearth+crot*vearth)
+  !!
+  !! @param[out] xlon dx/dlon in 1/degrees
+  !! @param[out] xlon dx/dlat in 1/degrees
+  !! @param[out] xlat dy/dlon in 1/degrees
+  !! @param[out] ylon dy/dlon in 1/degrees
+  !! @param[out] ylat dy/dlat in 1/degrees
+  !! @param[out] area Area weights in m^2.
+  !! Proportional to the square of the map factor in the case of
+  !! conformal projections
+  !!
+  !! @author George Gayno, Mark Iredell
+  !! @date July 2021
```

Comment:
OK, definitely you are using the date tag wrong. It should be the date from here:

`    !   PRGMMR: IREDELL       ORG: W/NMC23       DATE: 96-04-10
`